spec/controllers/online_applications_controller_spec.rb

```diff
@@ -157,7 +156,7 @@
       allow(application).to receive(:save)
       allow(ApplicationCalculation).to receive(:new).with(application).and_return(application_calculation)
       allow(ResolverService).to receive(:new).with(application, user).and_return(resolver_service)
-
+      allow(application.saving).to receive(:save)
```

Comment:
I think here it would be better only to test that the service is being called correctly, instead of actually calling it. Like the other services, you can do:

```ruby
  # define a dummy service
  let(:savings_service) { double(calculate!: nil) }

  # setup the real call to return the dummy service
  before do
    allow(SavingsPassFailService).to receive(:new).with(application.saving).and_return(savings_service)
  end

  # make sure the service has been called
  it 'runs the savings calculation' do
    expect(savings_service).to have_received(:calculate!)
  end 
```